src/nvim/options.lua

```diff
@@ -2395,6 +2400,8 @@ return {
       cb = 'did_set_eadirection',
       defaults = { if_true = 'both' },
       values = { 'both', 'ver', 'hor' },
+      -- TODO(lewis6991): #31944 removed the ability to set to empty. Is empty
+      -- allowed?
```

Comment:
Not allowed. Because it doesn't have a different behavior than the other values.

Whereas for example `buftype=''` has a (documented), different behavior.

Reply:
What about `fileformat`? We have a functional test that explicitly sets it to empty?

Reply:
Using Vim 9.1, `:let &fileformat=''` silently sets "dos", even though its default is "unix" on my system. That's a pretty cool feature but I think we can drop it :D

Reply:
> We have a functional test that explicitly sets it to empty.

Let's update the test to assert the new, proper behavior :P